src/zenml/zen_stores/base_zen_store.py

```diff
@@ -309,37 +299,19 @@ def get_default_store_config(path: str) -> StoreConfiguration:
 
     def _initialize_database(self) -> None:
         """Initialize the database on first use."""
-        try:
-            default_workspace = self._default_workspace
-        except KeyError:
-            default_workspace = self._create_default_workspace()
-        try:
-            assert self._admin_role
-        except KeyError:
-            self._create_admin_role()
-        try:
-            assert self._guest_role
-        except KeyError:
-            self._create_guest_role()
+        default_workspace = self._get_or_create_default_workspace()
 
         config = ServerConfiguration.get_server_config()
-        # If the auth scheme is external, don't create the default user and
-        # stack
+        # If the auth scheme is external, don't create the default user
         if config.auth_scheme != AuthScheme.EXTERNAL:
             try:
-                default_user = self._default_user
-            except KeyError:
-                default_user = self._create_default_user()
-            try:
-                self._get_default_stack(
-                    workspace_name_or_id=default_workspace.id,
-                    user_name_or_id=default_user.id,
-                )
+                _ = self._default_user
             except KeyError:
-                self._create_default_stack(
-                    workspace_name_or_id=default_workspace.id,
-                    user_name_or_id=default_user.id,
-                )
+                self._create_default_user()
+
+        self._get_or_create_default_stack(
```

Comment:
Given that the default workspace and stack are created during the store DB initialization which happens at server startup (or when the local DB is initialized on local deployment), I think we should only call `_get_or_create_default_stack` and `_get_or_create_default_workspace` here in this initialization block, but everywhere else we should call `_get_default_stack` and `_get_default_user` and just fail if they're not present. The reason is that I'm afraid to have these created ad-hoc past the point where the store is initialized now. This is especially relevant if the client calls these instead of the server.

I'm talking here mostly about the `validate_active_config` method which should no longer create default entities if they're not already there.

Reply:
I would even go as far as moving the creation of the default stack/user/workspace to the SQL zen store, where they actually belong.

Reply:
This is all just optional BTW, it works fine as it is.

Reply:
I've moved all the database initialization into the SQLZenStore as you suggested, much better :)